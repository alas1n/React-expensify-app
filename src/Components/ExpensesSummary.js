import React from 'react'
import { connect } from 'react-redux'
import SelectExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
	const expensePlural = expenseCount > 1 ? 'expenses' : 'expense'
	return (
		<div>
			<h1>You have {expenseCount} {expensePlural} total of : {numeral(expenseTotal / 100).format('$0,0.00')}</h1>
		</div>
	)
}



const mapStateToProps = (state) => {
	const visibleExpenses = SelectExpenses(state.expenses, state.filters)
	return {
		expenseCount: visibleExpenses.length,
		expenseTotal: getExpensesTotal(visibleExpenses)
	}
}


export default connect(mapStateToProps)(ExpensesSummary)
