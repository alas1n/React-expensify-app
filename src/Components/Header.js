import React from 'react'
import {NavLink } from 'react-router-dom'



const Header = () => (
    <div>
        <h1> Expense App </h1>
        <NavLink to="/" activeClassName="is-active" exact> home </NavLink> <br></br>
        <NavLink to="/create" activeClassName="is-active"> create </NavLink> <br></br>
        <NavLink to="/edit" activeClassName="is-active"> edit </NavLink> <br></br>
        <NavLink to="/help" activeClassName="is-active"> help </NavLink> <br></br>
    </div>
)

export default Header