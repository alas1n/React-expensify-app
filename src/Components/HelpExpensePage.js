import React from 'react'


const HelpExpensePage = () => (
    <div>
        this is from Help expense
    </div>
)

export default HelpExpensePage