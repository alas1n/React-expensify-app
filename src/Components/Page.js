import React from 'react'
import {Link} from 'react-router-dom'

const Page = () => (
    <div>
        404! - <Link to="/"> home </Link>
    </div>
)

export default Page