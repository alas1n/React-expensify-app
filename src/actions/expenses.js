import uuid from 'uuid'


// add expenses
export const addExpenses = (
    {
        description = '',
        note = '',
        amount = 0,
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSES',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
})

// REMOVE EXPENSES
export const removeExpenses = (({ id } = {}) => {
    return {
        type: 'REMOVE_EXPENSE',
        id
    }
})

// EDIT EXPENSES
export const editExpenses = ((id, updates) => {
    return {
        type: 'EDIT_EXPENSE',
        id,
        updates
    }
})