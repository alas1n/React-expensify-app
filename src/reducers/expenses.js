// DEFULT VALUE FOR EXPENSE ARRY
const expneseReducerDefaultState = [];


// EXPENSE REDUCER
export default (state = expneseReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSES':
            return [
                ...state,
                action.expense
            ]
        case 'REMOVE_EXPENSE':
            return state.filter(expense => expense.id !== action.id)
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                } else {
                    return expense
                }
            })
        default:
            return state
    }
}


