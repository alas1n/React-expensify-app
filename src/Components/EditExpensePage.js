import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { editExpenses, removeExpenses } from '../actions/expenses'



export class EditExpensePage extends React.Component {
    onSubmit = (expense) => {
        this.props.editExpenses(this.props.expense.id, expense)
        this.props.history.push('/')
    }
    onClick = () => {
        this.props.removeExpenses({ id: this.props.expense.id })
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}
                />
                <button
                    onClick={this.onClick}
                >Remove
                </button>
            </div>
        )
    }
}


const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)
    }
}

const mapDispatchToProps = (dispatch, props) => {
    // dont know why he put props
    return{
        editExpenses: (id, expense) => dispatch(editExpenses(id, expense)),
        removeExpenses: (id) => dispatch(removeExpenses(id))
    }

}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)



 