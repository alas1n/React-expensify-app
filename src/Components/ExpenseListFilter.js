import React from 'react'
import { connect } from 'react-redux'
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters'




export class ExpensesListFilters extends React.Component {
    state = {
        calendarFocused: null
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }))
    }
    setText = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    setFilter = (e) => {
        if (e.target.value === 'date') {
            this.props.sortByDate()
        } else if (e.target.value === 'amount') {
            this.props.sortByAmount()
        }
    }
    render() {
        return (
            <div>
                <input
                    type='text'
                    onChange={this.setText}
                />
                <select
                    onChange={this.setFilter}
                >
                    <option value='date'>Date</option>
                    <option value='amount'>Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        )
    }
}



const mapDispatchToProps = (dispatch) => {
    return {
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate)),
        setTextFilter: (textFilterDateOrAmount) => dispatch(setTextFilter(textFilterDateOrAmount)),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () => dispatch(sortByAmount())
    }
}



const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpensesListFilters);