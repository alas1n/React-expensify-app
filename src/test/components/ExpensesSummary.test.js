import React from 'react'
import {ExpensesSummary} from '../../Components/ExpensesSummary'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'

test('should render Expense Summary with one expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={100} />)
	expect(wrapper).toMatchSnapshot()
})


test('should render Expense Summary more than one expense', () => {
	const wrapper = shallow(<ExpensesSummary expenseCount={2} expenseTotal={0} />)
	expect(wrapper).toMatchSnapshot()
})

