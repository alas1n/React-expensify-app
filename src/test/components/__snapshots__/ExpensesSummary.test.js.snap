// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense Summary more than one expense 1`] = `
<div>
  <h1>
    You have 
    2
     
    expenses
     total of : 
    $0.00
  </h1>
</div>
`;

exports[`should render Expense Summary with one expense 1`] = `
<div>
  <h1>
    You have 
    1
     
    expense
     total of : 
    $1.00
  </h1>
</div>
`;
